client/lib/titan/has-titan-mail-with-us.js

```diff
@@ -1,4 +1,9 @@
-export function hasTitanMailWithUs() {
-	// TODO: make the actual check
-	return false;
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+export function hasTitanMailWithUs( domain ) {
+	const domainStatus = get( domain, 'titanMailSubscription.status', '' );
+	return domainStatus === 'active';
```

Comment:
In the future we should probably deal with alternate states for proper user notification.